backend/app/use_case/recommendation.py

```diff
@@ -10,40 +10,91 @@
 openai.api_base = "https://api.openai.iniad.org/api/v1"
 
 
+def extract_json(openai_response: str) -> str:
+    """
+    Extract json from openai response
+    """
+    json_start_pos = openai_response.find("{")
+    json_end_pos = openai_response.rfind("}")
+    json = openai_response[json_start_pos : json_end_pos + 1]
+    return json
```

Comment:
This one return string type instead of json so you need to convert into json first.

```python
import json

def extract_json(openai_response: str) -> str:
    """
    Extract json from openai response
    """
    json_start_pos = openai_response.find("{")
    json_end_pos = openai_response.rfind("}")
    data = openai_response[json_start_pos : json_end_pos + 1]
    return json.loads(data)
```